cpu/kinetis_common/periph/uart.c

```diff
@@ -64,32 +143,62 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
         return res;
     }
 
+    /* initialize and lock DMA mutex */
+    mutex_init(&_uart_dma_mutex[uart]);
+    mutex_lock(&_uart_dma_mutex[uart]);
+
     /* remember callback addresses */
     config[uart].rx_cb = rx_cb;
     config[uart].arg = arg;
 
-    /* enable receive interrupt */
+    /* enable interrupt source */
     switch (uart) {
 #if UART_0_EN
-
-        case UART_0:
+        case UART_DEV(0):
             NVIC_SetPriority(UART_0_IRQ_CHAN, UART_IRQ_PRIO);
```

Comment:
I think this should go in a separate PR which gets rid of the whole switch block